web/console/src/app/components/common/MatchFinder/index.tsx

```diff
@@ -67,41 +67,45 @@ const MatchFinder: React.FC = () => {
     /** Sends confirm action. */
     const confirmMatch = () => {
         setIsMatchConfirmed(true);
-        queueSendAction('confirm', squad.id);
+        sendAction('confirm', squad.id);
     };
 
     /** Canceles confirmation game. */
     const cancelConfirmationGame = () => {
-        queueSendAction('reject', squad.id);
+        sendAction('reject', squad.id);
     };
 
     /** Canceles searching game and closes MatchFinder component. */
     const canselSearchingGame = () => {
-        onOpenConnectionSendAction('finishSearch', squad.id);
+        setMatchQueue();
+
+        sendAction('finishSearch', squad.id);
 
         /** Updates current queue client. */
-        const updatedClient = getCurrentQueueClient();
-        setQueueClient(updatedClient);
+        const updatedClient = getCurrentWebSocketClient();
+        setWebSocketClient(updatedClient);
 
         dispatch(startSearchingMatch(false));
     };
 
     /** Exposes start searching match logic. */
     const startSearchMatch = () => {
-        onOpenConnectionSendAction('startSearch', squad.id);
+        setMatchQueue();
 
-        /** Updates current queue client. */
-        const newclient = getCurrentQueueClient();
-        setQueueClient(newclient);
+        sendAction('startSearch', squad.id);
+
+        /** Updates current websocket client. */
+        const newclient = getCurrentWebSocketClient();
+        setWebSocketClient(newclient);
     };
 
     useEffect(() => {
         isSearchingMatch && startSearchMatch();
     }, [isSearchingMatch]);
 
-    /** Processes queue client event messages. */
-    if (queueClient) {
-        queueClient.ws.onmessage = ({ data }: MessageEvent) => {
+    /** Processes websocket client event messages. */
+    if (webSocketClient) {
+        webSocketClient.ws.onmessage = ({ data }: MessageEvent) => {
```

Comment:
will it possbile place this logic to client level?

Reply:
I guess no, because it is logic related only to matches and there are setState functions, which are in MatchFinder component